sim/dex.ts

```diff
@@ -897,7 +912,31 @@ export class ModdedDex {
 			return rule.charAt(0) + this.validateBanRule(rule.slice(1));
 		default:
 			const id = toID(rule);
-			if (!this.data.Formats.hasOwnProperty(id)) {
+			const parts = rule.replace('=', ':').split(":");
+			if (parts.length > 1) {
+				const type = toID(parts[0]);
+				if (type === "teamlength" || type === "teamcap") {
+					let options = parts[1].split('-');
+					const onValidate = parseInt(options[0]);
+					let onBattle = parseInt(options[1]);
+					if (options.length === 1) onBattle = onValidate;
+					if (!isNaN(onValidate) && !isNaN(onBattle)) {
+						if (onValidate < onBattle) {
+							throw new Error("You can't bring more Pokémon to a battle than the maximum team length");
+						}
+						if (onValidate > 24) {
+							throw new Error("You may not have more than 24 Pokémon on a team");
+						}
+						return `teamlength:${onBattle}-${onValidate}`;
```

Comment:
It would probably make sense to return an array like `complexBan` and `complexTeamBan`, instead of needing to parse this twice?

Reply:
The return value is added to the format string so needs to return a string unless I'm really stupid

Reply:
```ts
			const ruleSpec = this.validateRule(rule, format);
			if (typeof ruleSpec !== 'string') {
```

Look at line 810: there's a test for the return value (for `complexBan` etc).

Reply:
I'm confused why you resolved this. This doesn't seem to have happened at all? You're still returning a string here.

Reply:
You're absolutely correct, let me check how it's possible because I could've sworn I changed that

Reply:
Apologies, I did indeed forget to commit that edit